pkg/gofr/datasource/clickhouse/clickhouse.go

```diff
@@ -7,77 +7,76 @@ import (
 	"strings"
 	"time"
 
+	"github.com/ClickHouse/clickhouse-go/v2"
 	"go.opentelemetry.io/otel/attribute"
-
 	"go.opentelemetry.io/otel/trace"
-
-	"github.com/ClickHouse/clickhouse-go/v2"
+	
 )
 
+
+// Config defines the configuration needed to connect to a Clickhouse database.
 type Config struct {
-	Hosts    string
-	Username string
-	Password string
-	Database string
+	Hosts    string // Comma-separated list of Clickhouse host addresses.
+	Username string // Username for authentication.
+	Password string // Password for authentication.
+	Database string // Target database name.
 }
 
+// Client represents the Clickhouse client, encapsulating connection details, configuration, logging, metrics, and tracing.
 type Client struct {
-	conn    Conn
-	config  Config
-	logger  Logger
-	metrics Metrics
-	tracer  trace.Tracer
+	conn    Conn        // The database connection interface.
+	config  Config      // Configuration for the Clickhouse connection.
+	logger  Logger      // Logger interface for logging messages.
+	metrics Metrics     // Metrics interface for monitoring database operations.
+	tracer  trace.Tracer // OpenTelemetry tracer for distributed tracing.
 }
 
-var errStatusDown = errors.New("status down")
-
-// New initializes Clickhouse client with the provided configuration.
-// Metrics, Logger has to be initialized before calling the Connect method.
-// Usage:
-//
-//	client.UseLogger(Logger())
-//	client.UseMetrics(Metrics())
-//
-//	client.Connect()
+// New initializes a new Clickhouse client with the provided configuration.
+// Metrics and logger must be set separately before calling the Connect method.
 func New(config Config) *Client {
 	return &Client{config: config}
 }
 
 // UseLogger sets the logger for the Clickhouse client.
+// The logger must implement the Logger interface.
 func (c *Client) UseLogger(logger interface{}) {
 	if l, ok := logger.(Logger); ok {
 		c.logger = l
 	}
 }
 
 // UseMetrics sets the metrics for the Clickhouse client.
+// The metrics must implement the Metrics interface.
```

Comment:
```suggestion
// The metrics must implement the [Metrics] interface.
```

And same remark about "must"